packages/@lwc/ssr-compiler/src/compile-template/component.ts

```diff
@@ -75,22 +77,36 @@ function reflectAriaPropsAsAttrs(props: IrProperty[]): IrAttribute[] {
         .filter((el): el is NonNullable<IrAttribute> => el !== null);
 }
 
-export const Component: Transformer<IrComponent> = function Component(node, cxt) {
-    // Import the custom component's generateMarkup export.
-    const childGeneratorLocalName = `generateMarkup_${toPropertyName(node.name)}`;
-    const importPath = kebabcaseToCamelcase(node.name);
-    const componentImport = bImportGenerateMarkup(childGeneratorLocalName, importPath);
-    cxt.hoist(componentImport, childGeneratorLocalName);
-    const childTagName = node.name;
-
+export function ImportedComponent(
+    node: IrComponent,
+    childGenerator: Expression,
+    tagName: Expression,
+    cxt: TransformerContext
+): EsStatement[] {
     const attributes = [...node.attributes, ...reflectAriaPropsAsAttrs(node.properties)];
 
     return [
         bYieldFromChildGenerator(
             getChildAttrsOrProps(node.properties, cxt),
             getChildAttrsOrProps(attributes, cxt),
-            b.identifier(childGeneratorLocalName),
-            b.literal(childTagName)
+            childGenerator,
+            tagName
         ),
     ];
+}
+
+export const Component: Transformer<IrComponent> = function Component(node, cxt) {
+    // Import the custom component's generateMarkup export.
+    const childGeneratorLocalName = `generateMarkup_${toPropertyName(node.name)}`;
```

Comment:
I think `node.name` can have hyphens in it here. If so, that would be an invalid JS identifier, so we'll need to transform it into something compliant.

Reply:
Changed it to use a uid.

Reply:
And added a test for a component that has two child components.